File: srcs/yeet_text.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   yeet_text.c                                        :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: becaraya <becaraya@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2019/12/12 23:36:45 by becaraya          #+#    #+#             */
/*   Updated: 2019/12/13 21:33:36 by becaraya         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "doom_nukem.h"

static void		ft_rectdel(SDL_Rect **re)
{
	ft_memdel((void **)re);
}

void			free_text_next(t_al *al)
{
	al->text.sector.str ? ft_strdel(&al->text.sector.str) : 0;
	al->text.sector.where ? ft_rectdel(&al->text.sector.where) : 0;
	al->text.tools.str ? ft_strdel(&al->text.tools.str) : 0;
	al->text.tools.where ? ft_rectdel(&al->text.tools.where) : 0;
	al->text.set_spawn.str ? ft_strdel(&al->text.set_spawn.str) : 0;
	al->text.set_spawn.where ? ft_rectdel(&al->text.set_spawn.where) : 0;
	al->text.set_bad_pig.str ? ft_strdel(&al->text.set_bad_pig.str) : 0;
	al->text.set_bad_pig.where ? ft_rectdel(&al->text.set_bad_pig.where) : 0;
	al->text.wall_tex.str ? ft_strdel(&al->text.wall_tex.str) : 0;
	al->text.wall_tex.where ? ft_rectdel(&al->text.wall_tex.where) : 0;
	al->text.fl_tex.str ? ft_strdel(&al->text.fl_tex.str) : 0;
	al->text.fl_tex.where ? ft_rectdel(&al->text.fl_tex.where) : 0;
	al->text.ce_tex.str ? ft_strdel(&al->text.ce_tex.str) : 0;
	al->text.ce_tex.where ? ft_rectdel(&al->text.ce_tex.where) : 0;
	al->text.fl_hei.str ? ft_strdel(&al->text.fl_hei.str) : 0;
	al->text.fl_hei.where ? ft_rectdel(&al->text.fl_hei.where) : 0;
	al->text.ce_hei.str ? ft_strdel(&al->text.ce_hei.str) : 0;
	al->text.ce_hei.where ? ft_rectdel(&al->text.ce_hei.where) : 0;
	al->text.draw.str ? ft_strdel(&al->text.draw.str) : 0;
	al->text.draw.where ? ft_rectdel(&al->text.draw.where) : 0;
	al->text.link.str ? ft_strdel(&al->text.link.str) : 0;
	al->text.link.where ? ft_rectdel(&al->text.link.where) : 0;
}

void			free_text(t_al *al)
{
	al->text.gen_map.str ? ft_strdel(&al->text.gen_map.str) : 0;
	al->text.gen_map.where ? ft_rectdel(&al->text.gen_map.where) : 0;
	al->text.sect_index.str ? ft_strdel(&al->text.sect_index.str) : 0;
	al->text.sect_index.where ? ft_rectdel(&al->text.sect_index.where) : 0;
	al->text.wall_index.str ? ft_strdel(&al->text.wall_index.str) : 0;
	al->text.wall_index.where ? ft_rectdel(&al->text.wall_index.where) : 0;
	al->text.x_start.str ? ft_strdel(&al->text.x_start.str) : 0;
	al->text.x_start.where ? ft_rectdel(&al->text.x_start.where) : 0;
	al->text.y_start.str ? ft_strdel(&al->text.y_start.str) : 0;
	al->text.y_start.where ? ft_rectdel(&al->text.y_start.where) : 0;
	al->text.x_end.str ? ft_strdel(&al->text.x_end.str) : 0;
	al->text.x_end.where ? ft_rectdel(&al->text.x_end.where) : 0;
	al->text.y_end.str ? ft_strdel(&al->text.y_end.str) : 0;
	al->text.y_end.where ? ft_rectdel(&al->text.y_end.where) : 0;
	al->text.y_end.str ? ft_strdel(&al->text.cancel.str) : 0;
	al->text.y_end.where ? ft_rectdel(&al->text.cancel.where) : 0;
	al->text.settings.str ? ft_strdel(&al->text.cancel.str) : 0;
	al->text.settings.where ? ft_rectdel(&al->text.cancel.where) : 0;
	al->text.wall.str ? ft_strdel(&al->text.wall.str) : 0;
	al->text.wall.where ? ft_rectdel(&al->text.wall.where) : 0;
	free_text_next(al);
}

File: web version/src/yeet_text.js
function freeRect(rect) {
    rect = null;
}

function freeTextHeiNum(al) {
    al.text.ceHeiNum.str = null;
    al.text.ceHeiNum.where = null;
    al.text.flHeiNum.str = null;
    al.text.flHeiNum.where = null;
}

function freeTextNextNext(al) {
    al.text.setPlayer.str = null;
    al.text.setPlayer.where = null;
    al.text.life.str = null;
    al.text.life.where = null;
    al.text.power.str = null;
    al.text.power.where = null;
    al.text.weapon.str = null;
    al.text.weapon.where = null;
    al.text.size.str = null;
    al.text.size.where = null;
    al.text.mass.str = null;
    al.text.mass.where = null;
    al.text.gravity.str = null;
    al.text.gravity.where = null;
    al.text.isDoor.str = null;
    al.text.isDoor.where = null;
}

function freeTextNext(al) {
    al.text.sector.str = null;
    al.text.sector.where = null;
    al.text.tools.str = null;
    al.text.tools.where = null;
    al.text.setSpawn.str = null;
    al.text.setSpawn.where = null;
    al.text.setEnd.str = null;
    al.text.setEnd.where = null;
    al.text.setBadPig.str = null;
    al.text.setBadPig.where = null;
    al.text.wallTex.str = null;
    al.text.wallTex.where = null;
    al.text.flTex.str = null;
    al.text.flTex.where = null;
    al.text.ceTex.str = null;
    al.text.ceTex.where = null;
    al.text.flHei.str = null;
    al.text.flHei.where = null;
    al.text.ceHei.str = null;
    al.text.ceHei.where = null;
    al.text.draw.str = null;
    al.text.draw.where = null;
    al.text.link.str = null;
    al.text.link.where = null;
}

function freeText(al) {
    al.text.genMap.str = null;
    al.text.genMap.where = null;
    al.text.sectIndex.str = null;
    al.text.sectIndex.where = null;
    al.text.wallIndex.str = null;
    al.text.wallIndex.where = null;
    al.text.settings.str = null;
    al.text.settings.where = null;
    al.text.wall.str = null;
    al.text.wall.where = null;
    al.text.resetMap.str = null;
    al.text.resetMap.where = null;
    al.text.resetPlayer.str = null;
    al.text.resetPlayer.where = null;

    freeTextHeiNum(al);
    freeTextNext(al);
    freeTextNextNext(al);
}